src/error.rs

```diff
@@ -156,6 +167,20 @@ pub enum ErrorKind {
     /// defined for a field, but the corresponding resolver is not provided.
     ResolverNotFound(String, String),
 
+    /// This error is returned if the root node could not be determined from the schema.
+    /// Note: This error should never be thrown. This is a critical error. If you see it,
+    /// please report it to the warpgrapher team.
+    MissingRootNode,
```

Comment:
Per the comments on the engine.rs file, I'd recommend looking at ways to get rid of this as a possibility.